jep/0000/README.adoc

```diff
@@ -0,0 +1,296 @@
+| Discussions-To
+| link:https://gitter.im/jenkinsci/external-fingerprint-storage[Gitter channel]
+//
+//
+// Uncomment if this JEP depends on one or more other JEPs.
+//| Requires
+//| :bulb: JEP-NUMBER, JEP-NUMBER... :bulb:
+//
+//
+// Uncomment and fill if this JEP is rendered obsolete by a later JEP
+//| Superseded-By
+//| :bulb: JEP-NUMBER :bulb:
+//
+//
+// Uncomment when this JEP status is set to Accepted, Rejected or Withdrawn.
+//| Resolution
+//| :bulb: Link to relevant post in the jenkinsci-dev@ mailing list archives :bulb:
+
+|===
+
+== Abstract
+
+This proposal extends Jenkins core to support storing of fingerprints in an external storage, and allow them to be tracked across Jenkins instances through the entire CI/CD flow.
+These external storages will be configurable by storage specific plugins.
+This JEP builds such a reference implementation for Redis.
+
+== Specification
+
+=== Top-level requirements
+
+* Fingerprint Storage has a pluggable architecture
+* Reference implementations:
+
+** Local XML-based file system storage (current implementation)
+** Redis backed Fingerprint Storage Plugin
+
+=== Design decisions
+
+This section contains a top-level description of concerns we need to address in the design.
+
+==== Fingerprint Storage destinations
+
+The solution should be generic enough in order to support common storage types. The following storage types are supported:
+
+* Local XML-based file system storage (current implementation)
+* Redis backed Fingerprint Storage Plugin
+
+The API does not force a particular ORM mapping for the storages.
+The storage plugins are passed on the `Fingerprint` objects, leaving it up to them to decide how to save the fingerprints.
+
+==== Instance Specificness of Fingerprints
+
+[NOTE]
+====
+In this proposal, fingerprint's unique ID (or ID) is used to refer (as of now) to the MD5 checksum of the fingerprinted file.
+====
+
+We propose a design where it is the responsibility of the plugin to provide instance specific fingerprints to Jenkins core when required for traditional read/write/update.
+This means that fingerprints stored in the storage with the same ID by two different Jenkins instances will not be merged.
+
+The idea behind this is that the external storage will work as a common store for all the fingerprints, but the intended design of current offering isn’t affected.
+
+However, the API includes functionality to retrieve fingerprints from other instances in the case when they are required for tracing purposes.
+This allows for tracing fingerprints across Jenkins instances.
+Implementation wise, this is achieved by plugins ensuring that Fingerprints have an associated Jenkins Instance ID.
+
+==== Local to External Migration
+
+When a plugin is configured and enabled, the fingerprints are transferred from the local storage to the external storage.
+
+This by followed by clearing the local database of fingerprints from Jenkins HOME ONLY if the writes are successful.
```

Comment:
This is followed?

Reply:
Hi @afalko, thanks for reviewing. No this is not followed yet, but according to JEP's guidelines, we need to frame it in the present tense as if it is already done [[source](https://github.com/jenkinsci/jep/tree/master/jep-template/0000)].